src/core/ledger/transaction.ts

```diff
@@ -211,24 +212,27 @@ export const addOnUsPayment = async ({
   metadata,
   payerUser,
   payeeUser,
-  memoPayer,
+  memoFromPayer,
   shareMemoWithPayee,
   lastPrice,
 }: IAddTransactionOnUsPayment) => {
   const dealerPath = await dealerAccountPath()
 
   const entry = MainBook.entry(description)
 
+  // This is the current workaround to block spam memos
+  const memoSharingCheck = shareMemoWithPayee && sats >= MEMO_SHARING_SATS_THRESHOLD
+
   entry
     .credit(accountPath(payeeUser._id), sats * payeeUser.ratioBtc, {
       ...metadata,
-      memoPayer: shareMemoWithPayee ? memoPayer : null,
+      memoFromPayer: memoSharingCheck ? memoFromPayer : undefined,
```

Comment:
Why is `undefined` preferred to `null`?

Reply:
this must not be part of the ledger, please do the validation previous to call this method

Reply:
also I think is better to do the filter in the query, if we change the threshold later all the memos will be lost

Reply:
> Why is `undefined` preferred to `null`?

@samerbuna preferred `undefined` in this scenario

Reply:
> also I think is better to do the filter in the query, if we change the threshold later all the memos will be lost

If I'm understanding you correctly @dolcalmi: I would not necessarily do your suggestion here.

I guess it would be more robust to filter on the fly but it would basically change the memo after the fact as perceived by the user. I think we can start with either share/no share the memo at transaction time based on a sats threshold, and that way we make it more "immutable" entry. 

if we could filter it after the fact by changing the threshold, then memo could disappear from the user wallet and that is indeed not a good thing I believe

Reply:
query = get transactions in user wallet https://github.com/GaloyMoney/galoy/blob/main/src/core/user-wallet.ts#L119 (we are already doing some mapping there)

Reply:
The latest two commits revert the original `addInvoice` method and tries to implement it in that `user-wallet` instead, but I'm not sure that it comprehensively catches things (like can a transaction get to the end user through another pathway that isn't `getTransactions`?).

I'm leaning to undoing those last two commits and putting things back to the original state of the PR with the `addInvoice` changes.

Reply:
**Update:** Confirmed with @dolcalmi that `getTransactions` is the only entrypoint. Based on discussion on Dev Call we decided to go ahead with these current changes